lint/rule_name.go

```diff
@@ -29,12 +30,41 @@ const nameSeparator string = "::"
 
 var ruleNameValidator = regexp.MustCompile("^([a-z0-9][a-z0-9-]*(::[a-z0-9][a-z0-9-]*)?)+$")
 
-// NewRuleName creates a RuleName from segments.
+// NewRuleName creates a RuleName from an AIP number and a unique name within
+// that AIP.
+func NewRuleName(aip int, name string) RuleName {
+	// Determine the group.
+	group := ""
+	if aip > 0 && aip < 1000 {
+		group = "core"
+	}
+
+	// Sanity check: If the group does not exist, complain.
+	if group == "" {
+		panic("Invalid AIP; no available group.")
+	}
```

Comment:
I think it is more what the linter-linter should check. It is a mistake by the rule developer, and should not panic to the users.

Reply:
This is a compile time check though. The linter will fail to _build_. It will never fail to execute once built.

Reply:
I see. Can you print the invalid AIP and the name to help us locate the error? Thanks!

Reply:
I'm confused how this could be a compile time check?

Reply:
I think it would be caught/panic when running tests.